pringles/backends/web_display.py

```diff
@@ -0,0 +1,104 @@
+import tornado.web
+import tornado.ioloop
+from tornado.ioloop import PeriodicCallback
+import os
+from pathlib import Path
+import threading
+import asyncio
+
+from pringles.models import Model
+
+
+class ServerThread(threading.Thread):
+
+    _join_called = False
+
+    def run(self):
+        new_io_loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(new_io_loop)
+        WebApplication.initialize()
+
+        ioloop = tornado.ioloop.IOLoop.current()
+
+        def healthcheck():
+            if ServerThread._join_called:
+                ioloop.stop()
+
+        # Add a healthcheck every 1 second
+        PeriodicCallback(healthcheck, 1000).start()
+
+        ioloop.start()
+        ioloop.close()
+
+    # Overriding join method to mark the ioloop to be disposed
+    def join(self):
+        ServerThread._join_called = True
+        super().join()
+
+
+web_model_display_thread = ServerThread()
+
+
+class WebApplication(tornado.web.Application):
+
+    initialized = False
+    url_prefix = ''
+    # Hacky way to force the ipython render method to wait for the tornado server to start
+    started_latch = threading.Semaphore(0)
+
+    # Display model as html request handler
+    class DisplayModel(tornado.web.RequestHandler):
+        def get(self):
+            self.write("Not implemented yet")
+
+    class TestGetWithBodyHandler(tornado.web.RequestHandler):
+
+        def get(self):
+            written_response = "holis"
+            self.write(written_response)
+
+    def __init__(self, url_prefix: str = ''):
+        super().__init__([
+            (url_prefix + r'/test', self.TestGetWithBodyHandler),
```

Comment:
This shouldn't be here

Reply:
Yes. That need to be there. 
When you start a tornado web app, you need to call the parent constructor, which is responsible for registering each handler you want.